userweb/app/modules/commonView/headerCtrl.js

```diff
@@ -46,7 +46,23 @@ angular
             $scope.userAgent = localStorage.getItem('userAgent');
             $scope.ipRegistered = localStorage.getItem('ipRegistered');
         }
-        
+
+        var gsQRCode = localStorage.getItem('gsQRCode');
+
+        if (gsQRCode == 1) {
+            var terminalName = localStorage.getItem('terminalName');
+            var terminalPassword = localStorage.getItem('terminalPassword');
+            var extensionNumber = localStorage.getItem('extensionNumber');
+            var companyName = localStorage.getItem('companyName');
+            var companyDomain = localStorage.getItem('companyDomain');
+            var userName = localStorage.getItem('userName');
+
+            $scope.gsQRCode = true;
+            $scope.QRCode = '<?xml version="1.0" encoding="utf-8"?><AccountConfig version="1"><Account><RegisterServer>' + companyDomain + '</RegisterServer><UserID>' + terminalName + '</UserID><AuthID>' + terminalName + '</AuthID><AuthPass>' + terminalPassword + '</AuthPass><AccountName>' + extensionNumber + ' ' + companyName + '</AccountName><DisplayName>' + userName + '</DisplayName><Voicemail>*93</Voicemail></Account></AccountConfig>';
```

Comment:
This also assumes the Voiceimail code is `*93` which can be configured per company basis.
We can create a Company method called getCompanyServiceCode in [its model](https://github.com/irontec/ivozprovider/blob/bleeding/library/IvozProvider/Model/Companies.php#L352) that returns the code related to a service so you can use `$voicemailExten = '*' . $company->getCompanyServiceCode('Voicemail');`